### DISCUSSION TITLE
max connection and number of connected connections

### DISCUSSION THREAD

**Discussion by Agwave** [sqlalchemy/sqlalchemy#7701] [Usage Questions]
Is there a way to use sqlachemy to obtain the maximum number of connections to the current database and the number of connections already connected

**Comment by zzzeek** [comment]
there's no generalized answer to this, it depends on what type of database you're targeting.  For example on MySQL can write queries against the server variables like this:

```
MariaDB [(none)]> select @@max_connections;
+-------------------+
| @@max_connections |
+-------------------+
|               151 |
+-------------------+
1 row in set (0.001 sec)
```

so in SQLAlchemy

```py
max_connections = connection.scalar(text("select @@max_connections"))
```
